src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,92 @@
-export const PeopleFilters = () => {
+import { Link, useSearchParams } from 'react-router-dom';
+import { Person } from '../types/Person';
+import React, { useState, useEffect } from 'react';
+
+interface PeopleFiltersProps {
+  persons: Person[];
+  onFilterChange: (filter: 'all' | 'male' | 'female') => void;
+  onCenturyChange: (century: string | 'all' | null) => void;
+  onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const PeopleFilters: React.FC<PeopleFiltersProps> = ({
+  onFilterChange,
+  onCenturyChange,
+  onSearchChange,
+}) => {
+  const [filter, setFilter] = useState<'all' | 'male' | 'female'>('all');
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [searchParams, setSearchParams] = useSearchParams();
```

Comment:
The use of `getAll` here might be misleading since it returns an array, but you are treating `centuryQuery` as if it could be a single value. Consider using `get` if you expect a single value or handle the array appropriately.